warehouse/localdisk.go

```diff
@@ -0,0 +1,125 @@
+package warehouse
+
+import (
+	"fmt"
+	"github.com/fullstorydev/hauser/config"
+	"github.com/nishanths/fullstory"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+type LocalDisk struct {
+	conf *config.Config
+}
+
+const (
+	timestampFile string = ".sync.hs"
+)
+
+var _ Warehouse = (*LocalDisk)(nil)
+
+func NewLocalDisk(c *config.Config) *LocalDisk {
+	if c.Local.UseStartTime {
+		filename := filepath.Join(c.Local.SaveDir, timestampFile)
+		if _, err := os.Stat(filename); !os.IsNotExist(err) {
+			os.Remove(filename)
+		}
+	}
+
+	return &LocalDisk{
+		conf: c,
+	}
+}
+
+// Currently returns the names of all the fields exported by BundleFields
+func (w *LocalDisk) GetExportTableColumns() []string {
+	allFields := BundleFields()
+	fieldNames := make([]string, 0, len(allFields))
+	for name := range allFields {
+		fieldNames = append(fieldNames, name)
+	}
+	return fieldNames
+}
+
+// Reads the last sync date from file
+func (w *LocalDisk) LastSyncPoint() (time.Time, error) {
+	t := beginningOfTime
+	if w.conf.Local.UseStartTime {
+		t = w.conf.Local.StartTime
+	}
+	if _, err := os.Stat(w.conf.Local.SaveDir); os.IsNotExist(err) {
+		panic(err)
+	}
+	filename := filepath.Join(w.conf.Local.SaveDir, timestampFile)
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		return t, nil
+	}
+	timebytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return t, err
+	}
+	synctime, err := time.Parse(time.RFC3339, string(timebytes))
+	return synctime, err
+}
+
+// Writes the current time to timestamp file
+func (w *LocalDisk) SaveSyncPoints(bundles ...fullstory.ExportMeta) error {
+	t := time.Now().UTC().Format(time.RFC3339)
```

Comment:
Good to know!

Reply:
Can we assume that the timestamps in bundles are already in UTC? Or should we convert them?

Reply:
The FullStory client we're using ultimately creates the timestamps using the `time.Unix` function: https://github.com/fullstorydev/hauser/blob/87d1bc348c6cbf5e5d39db84cd1bb80bcdc86c2d/vendor/github.com/nishanths/fullstory/fullstory.go#L85

The [godoc](https://golang.org/pkg/time/#Unix) for that function says it returns a Local time, so I think you'll need to keep converting to UTC. Even if it were UTC, it might make sense to convert anyway, just on the off chance that the library we're using to fetch bundles changes its behavior. I haven't read all the code for `time.Time`, but my understanding is that the `Location` is for presentation only, so I don't think changing it changes the underlying representation, and thus doesn't incur any appreciable cost.